lib/rules/require-spaces-in-for-statement.js

```diff
@@ -58,22 +58,19 @@ module.exports.prototype = {
 
     check: function(file, errors) {
         file.iterateNodesByType('ForStatement', function(node) {
-            var prevToken;
             if (node.test) {
-                var testToken = file.getTokenByRangeStart(node.test.range[0]);
-                prevToken = file.getPrevToken(testToken);
+                var testToken = file.getFirstNodeToken(node.test);
                 errors.assert.whitespaceBetween({
-                    token: prevToken,
+                    token: file.getPrevToken(testToken),
```

Comment:
Right, only certain rules where it's common to have comment blocks in between (such as function arguments) should we consider them. But again, I think it's beyond scope of this pull request. @zxqfox, file a new issue for that please.